includes/internal/emails/class-email-generator.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * File containing the Email_Generator class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Internal\Emails;
+
+use function Sensei_WC_Paid_Courses\E2e_Test_Data\Helpers\create_post;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class Email_Generator
+ *
+ * @package Sensei\Internal\Emails
+ */
+class Email_Generator {
+
+	/**
+	 * Initialize the class and add hooks.
+	 *
+	 * @internal
+	 */
+	public function init(): void {
+		add_action( 'sensei_user_course_start', [ $this, 'student_started_course_mail_to_teacher' ], 10, 2 );
+	}
+
+	/**
+	 * Send email to teacher when a student starts a course.
+	 *
+	 * @param int $student_id The student ID.
+	 * @param int $course_id  The course ID.
+	 *
+	 * @access private
+	 */
+	public function student_started_course_mail_to_teacher( $student_id, $course_id ) {
+
+		$this->temp_create_email();
```

Comment:
Just want to understand what are we going to do here later when emails are created during installation or update?
I mean it is still possible that the user deleted the post somehow. Should we try to find the post and fail (how?) if the post wasn't found?

Reply:
Hope you've got the answer here https://github.com/Automattic/sensei/pull/6533#discussion_r1107437914 :)